### ISSUE TITLE
DoHeatmap: Scale Differs when group.by Changes

### ISSUE THREAD

**Issue by atly2000** [satijalab/seurat#6776]
Hello,

How does DoHeatmap define the color scale? I sometimes see the range vary, from -1 to 1, or -2 to 2.

I am using DoHeatmap to plot heatmaps for my Seurat Object.
I assigned two new identity types for this object: Condition 1 and 2.
Condition 1 has  three categories: "ex", "nonex," and "other."
Condition 2 has 50 categories:
- It also contains "nonex" and "other," 
- However, cells assigned "ex" have been assigned one of 48 new categories named "c_01" to "c_48" based on a certain criteria we are testing.

**Both heatmaps use the same gene list.** 
When I plot the Heatmaps and group by Condition 1, I find the average within "ex", "nonex", and "other." I then plot a heatmap with 3 columns. 
When I plot the Heatmaps and group by Condition 2, I find the average within each cell assigned "c_01" to "c_48", "nonex", and "other." I then plot a heatmap with 50 columns (2 for nonex and other, 48 for the "c_" condition). 

However, the color scale has shifted between the Heatmaps grouped by Condition 1 vs Condition 2. 
I can tell because the colors in the "other" column, which contains identical cells between Condition 1 and Condition 2 grouping, have changed.

How can I make the color scale in the Condition 2 heatmap match the Condition 1 heatmap?
The blue arrows show the two columns that I expect to show the same color scale as they contain the same cells. 
(Gene names and exact column labels have been omitted)
![image](https://user-images.githubusercontent.com/87032440/206767355-6f8ce09c-ba3f-4a77-a735-bf4e61caa7bc.png)

Left: Grouped by Condition 2
Right: Grouped by Condition 1

**Comment by longmanz** [comment]
Hi,
It seems that there is no direct way to control the scale of the color bar. But you may do it via ggplot2() to change the scale by yourself (since the heatmaps are generated by ggplot2). For example:

`DoHeatmap(pbmc_small) + scale_fill_gradientn(limits = c(-2, 2), colours = PurpleAndYellow(), na.value = "white")`

You can modify the limits = c( lower_limit, upper_limit) to any other values, based on the maximal and minimal expression values in your datasets.

**Comment by atly2000** [comment]
Thank you for explaining. I tried specifying the scale like this but still got the same results.
I believe the scale of the heatmaps changed because I changed the "group.by" identities when calculating AverageExpression.

The first heatmap:
average_of_anno1_indiv <- AverageExpression(object = time1,group.by="clonotype_status",return.seurat = TRUE)
Where clonotype_status has 50 identities.

The second heatmap:
average_of_anno1 <- AverageExpression(object = time1,group.by="CD8_status",return.seurat = TRUE)
Where CD8_status has 3 identities.

Could the difference in number of identities affect the values on the heatmap?
